src/domain/PoolAddresses.ts

```diff
@@ -1,18 +1,23 @@
-import { TypedMap, JSONValue } from '@graphprotocol/graph-ts'
+import { TypedMap, JSONValue, Address } from '@graphprotocol/graph-ts'
 import { PoolAddresses } from '../../generated/schema'
 
+function toLowerCaseAddress(addr: JSONValue): string {
+  return Address.fromHexString(addr.toString()).toHexString()
+}
+
 export function updatePoolAddresses(poolId: string, addresses: TypedMap<string, JSONValue>): PoolAddresses {
   let poolAddresses = new PoolAddresses(poolId)
-  poolAddresses.coordinator = addresses.get('COORDINATOR').toString()
-  poolAddresses.assessor = addresses.get('ASSESSOR').toString()
-  poolAddresses.shelf = addresses.get('SHELF').toString()
-  poolAddresses.pile = addresses.get('PILE').toString()
-  poolAddresses.feed = addresses.get('FEED').toString()
-  poolAddresses.reserve = addresses.get('RESERVE').toString()
-  poolAddresses.seniorToken = addresses.get('SENIOR_TOKEN').toString()
-  poolAddresses.juniorToken = addresses.get('JUNIOR_TOKEN').toString()
-  poolAddresses.seniorTranche = addresses.get('SENIOR_TRANCHE').toString()
-  poolAddresses.juniorTranche = addresses.get('JUNIOR_TRANCHE').toString()
+
+  poolAddresses.coordinator = toLowerCaseAddress(addresses.get('COORDINATOR'))
```

Comment:
Have you double checked all potential edge cases here? I'm asking because I previously ran into an issue where I added lowercasing to addresses in one place, but I failed to lowercase addresses in another place.  I took a quick glance and I think this isn't the case now because the `PoolAddresses` object is always lowercased, but it would be good to be sure of this.

Reply:
Actually, taking a better look at this now, wouldn't one case of this be the `isSystemAccount` method. This gets passed arbitrary addresses as `account`, which might or might not be lowercased.  Shouldn't we lowercase `account` in that method?

Reply:
Another case where I think this might go wrong is in the `handleSupplyOrder` method. You're comparing `tokenAddresses.seniorTranche == tranche`, but `tranche` is an input address, which might not be lowercased.

Reply:
Maybe instead of a lowercase metho on storage ops we could add a `equalAddress` method?